### PR TITLE
added ArrayBuffer examples

### PR DESCRIPTION
**Pull request by maboa**
I added some ArrayBuffer examples.

Notes:

1. Sticking to one console.log per example for now. Happy to change though.

2. I still need to add attributes in the code tags (ie class, data-line and data-char) I based my examples on array-concat.html (which doesn't have them) so just noticed that now.

3. I included a constructor example - not sure if this is needed (also difficult to know what to console.log without using byteLength which was used in the next example).

4. Wasn't quite sure what to do for get ArrayBuffer[@@species] so left out for now.

5. ArrayBuffer.transfer() not currently working. Not sure if it is supported.

### CODE REVIEW

File: live-examples/js-examples/arraybuffer-slice.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+<code id="static-js">// create an ArrayBuffer with a size in bytes
+  var buffer = new ArrayBuffer(16);
+
+  var int32View = new Int32Array(buffer); // create the view
+
+  for (var i = 0; i < int32View.length; i++) {
+    int32View[i] = i;
+  }
+
+  // 0,1,2,3
```
Comment (wbamberg) [inline live-examples/js-examples/arraybuffer-slice.html]: This comment was unclear to me

File: live-examples/js-examples/arraybuffer-slice.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<pre>
+<code id="static-js">// create an ArrayBuffer with a size in bytes
+  var buffer = new ArrayBuffer(16);
+
+  var int32View = new Int32Array(buffer); // create the view
+
+  for (var i = 0; i < int32View.length; i++) {
+    int32View[i] = i;
+  }
```
Comment (wbamberg) [inline live-examples/js-examples/arraybuffer-slice.html]: I don't have any bright ideas here really, but is there a more concise/simple way we can initialise this array? It feels to me like this is quite extraneous to the example...

File: live-examples/js-examples/arraybuffer-constructor.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<pre>
+<code id="static-js">// create an ArrayBuffer with a size in bytes (in this case 8)
+var buffer = new ArrayBuffer(8);
+
+console.log(buffer);
+
+// expected output: [object ArrayBuffer]
+</code>
```
Comment (wbamberg) [inline live-examples/js-examples/arraybuffer-constructor.html]: I think this example is worth having. It shows us a valid argument to pass. Logging byteLength would be fine here too I think.

**Comment by schalkneethling** [comment]
> I still need to add attributes in the code tags (ie class, data-line and data-char) I based my examples on array-concat.html (which doesn't have them) so just noticed that now.

@maboa This is no longer needed. This was used before when we were using CodeMirror. I need to update the README to reflect this.

File: site.json [hunk 4]
```diff
@@ -178,6 +178,41 @@
...
+            "type": "js"
+        },
+        "arraybufferSlice": {
+            "baseTmpl": "tmpl/live-js-tmpl.html",
+            "exampleCode": "live-examples/js-examples/arraybuffer-slice.html",
+            "fileName": "arraybuffer-slice.html",
+            "title": "JavaScript Demo: ArrayBuffer.slice()",
+            "type": "js"
+        },
+        "arraybufferTransfer": {
```
Comment (wbamberg) [inline site.json]: I think you need to remove this entry, otherwise `npm run build` throws an error.

**Comment by wbamberg** [comment]
Closing this, because these examples are in https://github.com/mdn/interactive-examples/pull/237.

**Pull request closed without merging**